cli/src/commands/git/push.rs

```diff
@@ -402,15 +405,33 @@ fn validate_commits_ready_to_push(
         if commit.has_conflict()? {
             reasons.push("it has conflicts");
         }
+        let mut exposition = vec![];
         if !args.allow_private && is_private(commit.id())? {
             reasons.push("it is private");
+
+            if args.explain_denied {
+                exposition.push(format!(
+                    "commit {}, description: '{}'",
+                    short_commit_hash(commit.id()),
+                    commit
+                        .description()
+                        .lines()
+                        .next()
+                        .expect("commit should have at least one line with private description")
+                ));
+            }
         }
         if !reasons.is_empty() {
-            return Err(user_error(format!(
+            let mut message = format!(
                 "Won't push commit {} since {}",
                 short_commit_hash(commit.id()),
                 reasons.join(" and ")
-            )));
+            );
+            if args.explain_denied {
+                message.push('\n');
+                message.push_str(&exposition.join("\n"));
+            }
+            return Err(user_error(message));
```

Comment:
I personally don't think we need a flag to enable detailed error message.

For formatting, we can use commit_summary_template and `error.added_formatted_hint_with()`. See `check_rewritable()` in cmd_util.rs for example.

Reply:
Much obliged, will revise tomorrow or the next day!

Reply:
I've adopted this feedback during my last push, thank you for teaching me about those features of the codebase!